open_feature/provider/no_op_provider.py

```diff
@@ -16,7 +17,7 @@ def get_boolean_details(
         self,
         key: str,
         default_value: bool,
-        evaluation_context: typing.Any = None,
+        evaluation_context: EvaluationContext = EvaluationContext(),
```

Comment:
This may suffer from the same issue that I commented about before around mutable default arguments.

Reply:
As per my comment above, we should change this to ` = None` and then instantiate the `EvaluationContext` object in the method itself.

Reply:
Changed to use option 3